SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -329,29 +330,39 @@ void ASpatialDebugger::DrawDebugLocalPlayer(UCanvas* Canvas)
 	}
 }
 
-const FColor& ASpatialDebugger::GetVirtualWorkerColor(const Worker_EntityId EntityId) const
+FColor ASpatialDebugger::GetVirtualWorkerColor(const Worker_EntityId EntityId) const
 {
 	check(NetDriver != nullptr && !NetDriver->IsServer());
-
 	const AuthorityIntent* AuthorityIntentComponent = NetDriver->StaticComponentView->GetComponentData<AuthorityIntent>(EntityId);
-	const int32 VirtualWorkerId = (AuthorityIntentComponent != nullptr) ? AuthorityIntentComponent->VirtualWorkerId : SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
-
-	if (VirtualWorkerId != SpatialConstants::INVALID_VIRTUAL_WORKER_ID &&
-		VirtualWorkerId < ServerTintColors.Num())
-	{
-		return ServerTintColors[VirtualWorkerId];
-	}
-	else
-	{
+	check(AuthorityIntentComponent != nullptr);
+	const int32 VirtualWorkerId = AuthorityIntentComponent->VirtualWorkerId;
+	const FString* PhysicalWorkerId = NetDriver->VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(VirtualWorkerId);
+	if (PhysicalWorkerId == nullptr) {
+		// This can happen if the client hasn't yet received the VirtualWorkerTranslator mapping
 		return InvalidServerTintColor;
 	}
+	return SpatialGDK::GetColourForWorkerId(*PhysicalWorkerId);
```

Comment:
In the schema for the translator and other places I've seen this as physical worker _name_ instead of id (this is also the preferred terminology in the runtime afaik). Could you rename it for consistency?